src/libraries/tests.proj

```diff
@@ -242,12 +242,7 @@
     <ProjectExclusions Include="$(MSBuildThisFileDirectory)*\tests\**\*.Tests.csproj" />
 
     <!-- Functional tests on devices have problems with return codes from mlaunch -->
-    <ProjectExclusions Include="$(RepoRoot)\src\tests\FunctionalTests\iOS\Device\**\*.Test.csproj" />
-  </ItemGroup>
-
-  <ItemGroup Condition="'$(TargetOS)' == 'iossimulator' and '$(TargetArchitecture)' == 'arm64' and '$(RunDisablediOSTests)' != 'true'">
-    <!-- Functional tests on arm64 simulator have problems with return codes from mlaunch -->
-    <ProjectExclusions Include="$(RepoRoot)\src\tests\FunctionalTests\iOS\Simulator\**\*.Test.csproj" />
+    <ProjectExclusions Include="$(RepoRoot)\src\tests\FunctionalTests\iOS\Device\**\*.Test.csproj" Exclude="$(RepoRoot)\src\tests\FunctionalTests\iOS\Device\**\iOS.Device.Aot-Llvm.Test.csproj" />
   </ItemGroup>
```

Comment:
1. Does this mean that functional test on iossimulator-arm64 now do not have a problem with mlaunch?
2. Why for ios-arm64 we only run `iOS.Device.Aot-Llvm.Test.csproj`?

Reply:
> * Does this mean that functional test on iossimulator-arm64 now do not have a problem with mlaunch?

Not sure what was the issue. There are two xharness commands for testing: `apple run` and `apple test`. The `apple run` command could be used with any app where an exit code is set, while the `apple test` the test command is compatible with AppleTestRunner (https://github.com/dotnet/xharness/blob/main/README.md#my-apple-unit-tests-are-not-running).

There is a change that switches to the `apple run` for functional tests, which may have resolved the issue.

> * Why for ios-arm64 we only run `iOS.Device.Aot-Llvm.Test.csproj`?

Initially, we wanted to run the sample app, but after a discussion in https://github.com/dotnet/runtime/pull/87773#issuecomment-1630543788 we decided to switch to functional tests. The `iOS.Device.Aot-Llvm.Test.csproj` is almost the same as the sample app, so we picked that one. However, we could try to enable additional functional tests.

Reply:
> Not sure what was the issue. There are two xharness commands for testing: apple run and apple test. The apple run command could be used with any app where an exit code is set, while the apple test the test command is compatible with AppleTestRunner (https://github.com/dotnet/xharness/blob/main/README.md#my-apple-unit-tests-are-not-running).

> There is a change that switches to the apple run for functional tests, which may have resolved the issue.

I don't mind enabling the tests on `iossimulator-arm64` if they are passing, I was just asking whether this was an oversight or intentional. In any case, I would just suggest to note down everything that this PR changes (or brings up) in the PR description, as it touches many different places.

Reply:
What do you say about `iOS.Device.Aot.Test.csproj`?